### PR TITLE
Merge main into feature/feed

### PR DESCRIPTION
**Pull request by spoorthipujariadobe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by PravinPK** [comment]
I see that the classes have been moved from  `....messaging/internal` folder to `...messaging` folder. Is that change intentional in feed branch. And may I know the reason for the change ?

otherwise 👍🏼 from my side. All the code related to push notification building looks good.

**Review by PravinPK — APPROVED** [review]

**Comment by spoorthipujariadobe** [comment]
> I see that the classes have been moved from `....messaging/internal` folder to `...messaging` folder. Is that change intentional in feed branch. And may I know the reason for the change ?
> 
> otherwise 👍🏼 from my side. All the code related to push notification building looks good.

I believe this is to align with our package hierarchy in Android 2.x and was intentional. @rymorale can you confirm?

File: Documentation/sources/enum-public-classes/messaging-push-payload.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ Provides the type of action which needs to be performed when push notification i
     public ActionType getActionType()
 ```
 #### Enum `ActionType`
-There are 5 types of action available. These are used to determine which action needs to be performed when push notification is clicked.
+There are 4 types of action available. These are used to determine which action needs to be performed when push notification is clicked.
 `DEEPLINK`, `WEBURL`, `DISMISS`, `OPENAPP`, and `NONE`
```
[the remark this answers is not in the document]
Comment (spoorthipujariadobe) [reply Documentation/sources/enum-public-classes/messaging-push-payload.md:81]: We can mark it deprecated in `2.3.0` and remove in `3.x`?

[the remark this answers is not in the document]
Comment (spoorthipujariadobe) [reply Documentation/sources/enum-public-classes/messaging-push-payload.md]: Deprecated the enum and removed from docs

**Pull request merged**